arm-insights/2015-04-01/swagger/insightsManagementClient.json

```diff
@@ -0,0 +1,2119 @@
+          },
+          {
+            "name": "ruleName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the rule."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2016-03-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request for a list of alert rule related incidents",
+            "schema": {
+              "$ref": "#/definitions/IncidentListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/microsoft.insights/alertrules/{name}": {
+      "put": {
+        "description": "Creates or updates an alert rule",
+        "tags": [
+          "Alert"
+        ],
+        "operationId": "AlertRules_CreateOrUpdate",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The rule to create or update."
+          },
+          {
+            "name": "api-version",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "default": "2016-03-01",
+            "description": "The api-version to send to the server"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RuleCreateOrUpdateParameters"
+            },
+            "description": "The rule to create or update."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create or update an alert rule"
+          },
+          "201": {
+            "description": "Created alert rule"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/microsoft.insights/alertrules/{ruleName}": {
+      "delete": {
+        "description": "Deletes an alert rule",
+        "tags": [
+          "Alert"
+        ],
+        "operationId": "AlertRules_Delete",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
```

Comment:
It would be nice if the commonly repeated parameters are defined once in the global parameters section with `"x-ms-parameter-location": "method"` and reference it in the operation. In that way one does not have to repeat the same definition all the time.